Pod/Classes/VOKCoreDataManager.h

```diff
@@ -285,13 +289,20 @@ typedef void(^VOKObjectIDsReturnBlock)(VOKArrayOfManagedObjectIDs *managedObject
                      completion:(nullable VOKObjectIDsReturnBlock)completion;
 
 /**
- Saves any temporary managed object context and merges those changes with the main managed object context in a thread-safe way.
+ Saves any temporary managed object context and merges those changes with the main managed object context asynchronously.
  This method is safe to call from any queue.
- @param context     The termporary context to save. Do not use this method to save the main context.
+ @param context     The temporary context to save.
  */
 - (void)saveAndMergeWithMainContext:(NSManagedObjectContext *)context;
 
 /**
+ Saves any temporary managed object context and merges those changes with the main managed object context synchronously.
```

Comment:
Similar concern here—it sounds like this method ties an action in one context to an action in another, tying their queues together and potentially risking a deadlock.